pkg/api/controller.go

```diff
@@ -2845,6 +2846,53 @@ func (c *Controller) logCommitsHelper(w http.ResponseWriter, r *http.Request, re
 	writeResponse(w, http.StatusOK, response)
 }
 
+func (c *Controller) HeadObject(w http.ResponseWriter, r *http.Request, repository string, ref string, params HeadObjectParams) {
+	if !c.authorize(w, r, permissions.Node{
+		Permission: permissions.Permission{
+			Action:   permissions.ReadObjectAction,
+			Resource: permissions.ObjectArn(repository, params.Path),
+		},
+	}) {
+		return
+	}
+	ctx := r.Context()
+	c.LogAction(ctx, "head_object", r, repository, ref, "")
+
+	// read the FS entry
+	entry, err := c.Catalog.GetEntry(ctx, repository, ref, params.Path, catalog.GetEntryParams{ReturnExpired: true})
```

Comment:
btw, not in this pr - but I don't think we currently supports `ReturnExpired: true`

Reply:
Agree. Let's open an issue?